diff --git a/third_party/alp/include/alp/encode.hpp b/third_party/alp/include/alp/encode.hpp
index 1e5f562..a7b3c9d 100644
--- a/third_party/alp/include/alp/encode.hpp
+++ b/third_party/alp/include/alp/encode.hpp
@@ -301,11 +301,11 @@
 
 #ifdef __AVX512F__
 		for (size_t i {0}; i < config::VECTOR_SIZE; i = i + 8) {
-			__m512d l            = _mm512_loadu_pd(tmp_dbl_arr + i);
+			__m512d l            = _mm512_loadu_pd(encoded_dbl_arr + i);
 			__m512d r            = _mm512_loadu_pd(input_vector + i);
-			__m512i index        = _mm512_loadu_pd(INDEX_ARR + i);
+			__m512i index        = _mm512_loadu_si512((__m512i const*)(INDEX_ARR + i));
 			auto    is_exception = _mm512_cmpneq_pd_mask(l, r);
-			_mm512_mask_compressstoreu_pd(tmp_index + exceptions_idx, is_exception, index);
+			_mm512_mask_compressstoreu_epi64(INDEX_ARR + exceptions_idx, is_exception, index);
 			exceptions_idx += LOOKUP_TABLE[is_exception];
 		}
 #else
@@ -381,11 +381,11 @@
 
 #ifdef __AVX512F__
 		for (size_t i {0}; i < config::VECTOR_SIZE; i = i + 16) {
-			__m512  l            = _mm512_loadu_ps(tmp_dbl_arr + i);
+			__m512  l            = _mm512_loadu_ps(encoded_dbl_arr + i);
 			__m512  r            = _mm512_loadu_ps(input_vector + i);
-			__m512i index        = _mm512_loadu_ps(INDEX_ARR + i);
+			__m512i index        = _mm512_loadu_si512((__m512i const*)(INDEX_ARR + i));
 			auto    is_exception = _mm512_cmpneq_ps_mask(l, r);
-			_mm512_mask_compressstoreu_ps(tmp_index + exceptions_idx, is_exception, index);
+			_mm512_mask_compressstoreu_epi64(INDEX_ARR + exceptions_idx, is_exception, index);
 			exceptions_idx += LOOKUP_TABLE[is_exception];
 		}
 #else
