diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -59,3 +59,26 @@
 install(TARGETS lbug lbug_shared)
 
+set(_lbug_targets_file "${CMAKE_CURRENT_BINARY_DIR}/unofficial-ladybugdb-ladybug-targets.cmake")
+file(WRITE "${_lbug_targets_file}" [=[
+get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_DIR}/../../.." ABSOLUTE)
+if(NOT TARGET unofficial::ladybugdb::lbug)
+  add_library(unofficial::ladybugdb::lbug STATIC IMPORTED)
+  set_target_properties(unofficial::ladybugdb::lbug PROPERTIES IMPORTED_LOCATION "${_IMPORT_PREFIX}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}lbug${CMAKE_STATIC_LIBRARY_SUFFIX}" INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include")
+endif()
+if(NOT TARGET unofficial::ladybugdb::lbug_shared)
+  add_library(unofficial::ladybugdb::lbug_shared SHARED IMPORTED)
+  set_target_properties(unofficial::ladybugdb::lbug_shared PROPERTIES IMPORTED_LOCATION "${_IMPORT_PREFIX}/lib/${CMAKE_SHARED_LIBRARY_PREFIX}lbug${CMAKE_SHARED_LIBRARY_SUFFIX}" INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include")
+endif()
+]=])
+
+set(_lbug_config_file "${CMAKE_CURRENT_BINARY_DIR}/unofficial-ladybugdb-ladybug-config.cmake")
+file(WRITE "${_lbug_config_file}" [=[
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-ladybugdb-ladybug-targets.cmake")
+]=])
+
+install(
+    FILES "${_lbug_config_file}" "${_lbug_targets_file}"
+    DESTINATION "share/unofficial-ladybugdb-ladybug"
+)
+
 if(${BUILD_SINGLE_FILE_HEADER})
         # Create a command to generate lbug.hpp, and then create a target that is
