diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4cfc995..cfedf13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,16 +55,16 @@ set_target_properties(tree-sitter-sql
                       SOVERSION "${TREE_SITTER_ABI_VERSION}.${PROJECT_VERSION_MAJOR}"
                       DEFINE_SYMBOL "")
 
+include(GNUInstallDirs)
+
 configure_file(bindings/c/tree-sitter-sql.pc.in
                "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-sql.pc" @ONLY)
 
-include(GNUInstallDirs)
-
 install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/bindings/c/tree_sitter"
         DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
         FILES_MATCHING PATTERN "*.h")
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/tree-sitter-sql.pc"
-        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/pkgconfig")
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 install(TARGETS tree-sitter-sql
         LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
 
