diff --git a/fsst12.h b/fsst12.h
--- a/fsst12.h
+++ b/fsst12.h
@@ -124,17 +124,17 @@ fsst_compress(
 /* Loads two codes out of the first 24 bits of the 32-bit value and writes their symbols to the output buffer. */
-inline void write_two_codes_aligned(const unsigned char *__restrict__ len, unsigned long*__restrict__ symbol,
+inline void write_two_codes_aligned(const unsigned char *__restrict__ len, unsigned long long*__restrict__ symbol,
                             unsigned char *__restrict__ strOut, unsigned long &posOut, const unsigned long size,
                             const unsigned int code) {
    const unsigned int code0 = code & 4095;
    const unsigned int code1 = (code >> 12) & 4095;
    unsigned char *__restrict__ src, *__restrict__ lim, *__restrict__ dst = strOut+posOut;
    for(lim=strOut+((posOut+len[code0])>size?size:posOut+len[code0]), src=(unsigned char*__restrict__) &symbol[code0]; dst < lim; dst++, src++) *dst = *src;
    posOut += len[code0];
    for(lim=strOut+((posOut+len[code1])>size?size:posOut+len[code1]), src=(unsigned char*__restrict__) &symbol[code1]; dst < lim; dst++, src++) *dst = *src;
    posOut += len[code1];
 }
 
 /* Loads one code out of the first 12 bits of the 16-bit value and writes its symbol to the output buffer. */
-inline void write_one_code_aligned(const unsigned char *__restrict__ len, unsigned long*__restrict__ symbol,
+inline void write_one_code_aligned(const unsigned char *__restrict__ len, unsigned long long*__restrict__ symbol,
                                unsigned char *__restrict__ strOut, unsigned long &posOut, const unsigned long size,
                                unsigned short code) {
    code &= 4095;
