--- a/src/db/sqlengine/antlr/gen/SQLLexer.cc
+++ b/src/db/sqlengine/antlr/gen/SQLLexer.cc
@@ -10,7 +10,7 @@
 using namespace antlr4;
 
 SQLLexer::SQLLexer(CharStream *input) : Lexer(input) {
-  _interpreter = new atn::LexerATNSimulator(this, _atn, _decisionToDFA,
+  _interpreter = new atn::LexerATNSimulator(this, *_atn, _decisionToDFA,
                                             _sharedContextCache);
 }
 
@@ -42,12 +42,12 @@
   return _vocabulary;
 }
 
-const std::vector<uint16_t> SQLLexer::getSerializedATN() const {
+antlr4::atn::SerializedATNView SQLLexer::getSerializedATN() const {
   return _serializedATN;
 }
 
 const atn::ATN &SQLLexer::getATN() const {
-  return _atn;
+  return *_atn;
 }
 
 
@@ -56,8 +56,8 @@
 atn::PredictionContextCache SQLLexer::_sharedContextCache;
 
 // We own the ATN which in turn owns the ATN states.
-atn::ATN SQLLexer::_atn;
-std::vector<uint16_t> SQLLexer::_serializedATN;
+std::unique_ptr<antlr4::atn::ATN> SQLLexer::_atn;
+std::vector<int32_t> SQLLexer::_serializedATN;
 
 std::vector<std::string> SQLLexer::_ruleNames = {"OR",
                                                  "AND",
@@ -246,7 +246,7 @@
   // This code could be in a static initializer lambda, but VS doesn't allow
   // access to private class members from there.
   for (size_t i = 0; i < _symbolicNames.size(); ++i) {
-    std::string name = _vocabulary.getLiteralName(i);
+    std::string name(_vocabulary.getLiteralName(i));
     if (name.empty()) {
       name = _vocabulary.getSymbolicName(i);
     }
@@ -684,10 +684,10 @@
   atn::ATNDeserializer deserializer;
   _atn = deserializer.deserialize(_serializedATN);
 
-  size_t count = _atn.getNumberOfDecisions();
+  size_t count = _atn->getNumberOfDecisions();
   _decisionToDFA.reserve(count);
   for (size_t i = 0; i < count; i++) {
-    _decisionToDFA.emplace_back(_atn.getDecisionState(i), i);
+    _decisionToDFA.emplace_back(_atn->getDecisionState(i), i);
   }
 }
 
--- a/src/db/sqlengine/antlr/gen/SQLLexer.h
+++ b/src/db/sqlengine/antlr/gen/SQLLexer.h
@@ -82,11 +82,11 @@
   virtual const std::vector<std::string> &getChannelNames() const override;
   virtual const std::vector<std::string> &getModeNames() const override;
   virtual const std::vector<std::string> &getTokenNames()
-      const override;  // deprecated, use vocabulary instead
-  virtual antlr4::dfa::Vocabulary &getVocabulary() const override;
+      const;  // deprecated, use vocabulary instead
+  virtual antlr4::dfa::Vocabulary &getVocabulary() const;
 
-  virtual const std::vector<uint16_t> getSerializedATN() const override;
-  virtual const antlr4::atn::ATN &getATN() const override;
+  virtual antlr4::atn::SerializedATNView getSerializedATN() const;
+  virtual const antlr4::atn::ATN &getATN() const;
 
  private:
   static std::vector<antlr4::dfa::DFA> _decisionToDFA;
@@ -99,8 +99,8 @@
   static std::vector<std::string> _literalNames;
   static std::vector<std::string> _symbolicNames;
   static antlr4::dfa::Vocabulary _vocabulary;
-  static antlr4::atn::ATN _atn;
-  static std::vector<uint16_t> _serializedATN;
+  static std::unique_ptr<antlr4::atn::ATN> _atn;
+  static std::vector<int32_t> _serializedATN;
 
 
   // Individual action functions triggered by action() above.
--- a/src/db/sqlengine/antlr/gen/SQLParser.cc
+++ b/src/db/sqlengine/antlr/gen/SQLParser.cc
@@ -11,7 +11,7 @@
 using namespace antlr4;
 
 SQLParser::SQLParser(TokenStream *input) : Parser(input) {
-  _interpreter = new atn::ParserATNSimulator(this, _atn, _decisionToDFA,
+  _interpreter = new atn::ParserATNSimulator(this, *_atn, _decisionToDFA,
                                              _sharedContextCache);
 }
 
@@ -3175,8 +3175,8 @@
 atn::PredictionContextCache SQLParser::_sharedContextCache;
 
 // We own the ATN which in turn owns the ATN states.
-atn::ATN SQLParser::_atn;
-std::vector<uint16_t> SQLParser::_serializedATN;
+std::unique_ptr<antlr4::atn::ATN> SQLParser::_atn;
+std::vector<int32_t> SQLParser::_serializedATN;
 
 std::vector<std::string> SQLParser::_ruleNames = {"swallow_to_semi",
                                                   "compilation_unit",
@@ -3336,7 +3336,7 @@
 
 SQLParser::Initializer::Initializer() {
   for (size_t i = 0; i < _symbolicNames.size(); ++i) {
-    std::string name = _vocabulary.getLiteralName(i);
+    std::string name(_vocabulary.getLiteralName(i));
     if (name.empty()) {
       name = _vocabulary.getSymbolicName(i);
     }
@@ -3664,10 +3664,10 @@
   atn::ATNDeserializer deserializer;
   _atn = deserializer.deserialize(_serializedATN);
 
-  size_t count = _atn.getNumberOfDecisions();
+  size_t count = _atn->getNumberOfDecisions();
   _decisionToDFA.reserve(count);
   for (size_t i = 0; i < count; i++) {
-    _decisionToDFA.emplace_back(_atn.getDecisionState(i), i);
+    _decisionToDFA.emplace_back(_atn->getDecisionState(i), i);
   }
 }
 
--- a/src/db/sqlengine/antlr/gen/SQLParser.h
+++ b/src/db/sqlengine/antlr/gen/SQLParser.h
@@ -119,13 +119,13 @@
 
   virtual std::string getGrammarFileName() const override;
   virtual const antlr4::atn::ATN &getATN() const override {
-    return _atn;
+    return *_atn;
   };
-  virtual const std::vector<std::string> &getTokenNames() const override {
+  virtual const std::vector<std::string> &getTokenNames() const {
     return _tokenNames;
   };  // deprecated: use vocabulary instead.
-  virtual const std::vector<std::string> &getRuleNames() const override;
-  virtual antlr4::dfa::Vocabulary &getVocabulary() const override;
+  virtual const std::vector<std::string> &getRuleNames() const;
+  virtual antlr4::dfa::Vocabulary &getVocabulary() const;
 
 
   class Swallow_to_semiContext;
@@ -798,8 +798,8 @@
   static std::vector<std::string> _literalNames;
   static std::vector<std::string> _symbolicNames;
   static antlr4::dfa::Vocabulary _vocabulary;
-  static antlr4::atn::ATN _atn;
-  static std::vector<uint16_t> _serializedATN;
+  static std::unique_ptr<antlr4::atn::ATN> _atn;
+  static std::vector<int32_t> _serializedATN;
 
 
   struct Initializer {
--- a/src/db/index/column/fts_column/gen/FtsParser.h
+++ b/src/db/index/column/fts_column/gen/FtsParser.h
@@ -50,13 +50,13 @@
 
   virtual std::string getGrammarFileName() const override;
   virtual const antlr4::atn::ATN &getATN() const override {
-    return _atn;
+    return *_atn;
   };
-  virtual const std::vector<std::string> &getTokenNames() const override {
+  virtual const std::vector<std::string> &getTokenNames() const {
     return _tokenNames;
   };  // deprecated: use vocabulary instead.
-  virtual const std::vector<std::string> &getRuleNames() const override;
-  virtual antlr4::dfa::Vocabulary &getVocabulary() const override;
+  virtual const std::vector<std::string> &getRuleNames() const;
+  virtual antlr4::dfa::Vocabulary &getVocabulary() const;
 
 
   class Fts_query_unitContext;
@@ -290,8 +290,8 @@
   static std::vector<std::string> _literalNames;
   static std::vector<std::string> _symbolicNames;
   static antlr4::dfa::Vocabulary _vocabulary;
-  static antlr4::atn::ATN _atn;
-  static std::vector<uint16_t> _serializedATN;
+  static std::unique_ptr<antlr4::atn::ATN> _atn;
+  static std::vector<int32_t> _serializedATN;
 
 
   struct Initializer {
--- a/src/db/index/column/fts_column/gen/FtsParser.cc
+++ b/src/db/index/column/fts_column/gen/FtsParser.cc
@@ -11,7 +11,7 @@
 using namespace antlr4;
 
 FtsParser::FtsParser(TokenStream *input) : Parser(input) {
-  _interpreter = new atn::ParserATNSimulator(this, _atn, _decisionToDFA,
+  _interpreter = new atn::ParserATNSimulator(this, *_atn, _decisionToDFA,
                                              _sharedContextCache);
 }
 
@@ -972,8 +972,8 @@
 atn::PredictionContextCache FtsParser::_sharedContextCache;
 
 // We own the ATN which in turn owns the ATN states.
-atn::ATN FtsParser::_atn;
-std::vector<uint16_t> FtsParser::_serializedATN;
+std::unique_ptr<antlr4::atn::ATN> FtsParser::_atn;
+std::vector<int32_t> FtsParser::_serializedATN;
 
 std::vector<std::string> FtsParser::_ruleNames = {
     "fts_query_unit", "fts_or_expr",      "fts_and_expr",     "fts_seq_expr",
@@ -994,7 +994,7 @@
 
 FtsParser::Initializer::Initializer() {
   for (size_t i = 0; i < _symbolicNames.size(); ++i) {
-    std::string name = _vocabulary.getLiteralName(i);
+    std::string name(_vocabulary.getLiteralName(i));
     if (name.empty()) {
       name = _vocabulary.getSymbolicName(i);
     }
@@ -1106,10 +1106,10 @@
   atn::ATNDeserializer deserializer;
   _atn = deserializer.deserialize(_serializedATN);
 
-  size_t count = _atn.getNumberOfDecisions();
+  size_t count = _atn->getNumberOfDecisions();
   _decisionToDFA.reserve(count);
   for (size_t i = 0; i < count; i++) {
-    _decisionToDFA.emplace_back(_atn.getDecisionState(i), i);
+    _decisionToDFA.emplace_back(_atn->getDecisionState(i), i);
   }
 }
 
--- a/src/db/index/column/fts_column/gen/FtsLexer.h
+++ b/src/db/index/column/fts_column/gen/FtsLexer.h
@@ -39,11 +39,11 @@
   virtual const std::vector<std::string> &getChannelNames() const override;
   virtual const std::vector<std::string> &getModeNames() const override;
   virtual const std::vector<std::string> &getTokenNames()
-      const override;  // deprecated, use vocabulary instead
-  virtual antlr4::dfa::Vocabulary &getVocabulary() const override;
+      const;  // deprecated, use vocabulary instead
+  virtual antlr4::dfa::Vocabulary &getVocabulary() const;
 
-  virtual const std::vector<uint16_t> getSerializedATN() const override;
-  virtual const antlr4::atn::ATN &getATN() const override;
+  virtual antlr4::atn::SerializedATNView getSerializedATN() const;
+  virtual const antlr4::atn::ATN &getATN() const;
 
  private:
   static std::vector<antlr4::dfa::DFA> _decisionToDFA;
@@ -56,8 +56,8 @@
   static std::vector<std::string> _literalNames;
   static std::vector<std::string> _symbolicNames;
   static antlr4::dfa::Vocabulary _vocabulary;
-  static antlr4::atn::ATN _atn;
-  static std::vector<uint16_t> _serializedATN;
+  static std::unique_ptr<antlr4::atn::ATN> _atn;
+  static std::vector<int32_t> _serializedATN;
 
 
   // Individual action functions triggered by action() above.
--- a/src/db/index/column/fts_column/gen/FtsLexer.cc
+++ b/src/db/index/column/fts_column/gen/FtsLexer.cc
@@ -10,7 +10,7 @@
 using namespace antlr4;
 
 FtsLexer::FtsLexer(CharStream *input) : Lexer(input) {
-  _interpreter = new atn::LexerATNSimulator(this, _atn, _decisionToDFA,
+  _interpreter = new atn::LexerATNSimulator(this, *_atn, _decisionToDFA,
                                             _sharedContextCache);
 }
 
@@ -42,12 +42,12 @@
   return _vocabulary;
 }
 
-const std::vector<uint16_t> FtsLexer::getSerializedATN() const {
+antlr4::atn::SerializedATNView FtsLexer::getSerializedATN() const {
   return _serializedATN;
 }
 
 const atn::ATN &FtsLexer::getATN() const {
-  return _atn;
+  return *_atn;
 }
 
 
@@ -56,8 +56,8 @@
 atn::PredictionContextCache FtsLexer::_sharedContextCache;
 
 // We own the ATN which in turn owns the ATN states.
-atn::ATN FtsLexer::_atn;
-std::vector<uint16_t> FtsLexer::_serializedATN;
+std::unique_ptr<antlr4::atn::ATN> FtsLexer::_atn;
+std::vector<int32_t> FtsLexer::_serializedATN;
 
 std::vector<std::string> FtsLexer::_ruleNames = {
     "OR",          "AND",          "NOT",      "PLUS_SIGN",  "MINUS_SIGN",
@@ -86,7 +86,7 @@
   // This code could be in a static initializer lambda, but VS doesn't allow
   // access to private class members from there.
   for (size_t i = 0; i < _symbolicNames.size(); ++i) {
-    std::string name = _vocabulary.getLiteralName(i);
+    std::string name(_vocabulary.getLiteralName(i));
     if (name.empty()) {
       name = _vocabulary.getSymbolicName(i);
     }
@@ -247,10 +247,10 @@
   atn::ATNDeserializer deserializer;
   _atn = deserializer.deserialize(_serializedATN);
 
-  size_t count = _atn.getNumberOfDecisions();
+  size_t count = _atn->getNumberOfDecisions();
   _decisionToDFA.reserve(count);
   for (size_t i = 0; i < count; i++) {
-    _decisionToDFA.emplace_back(_atn.getDecisionState(i), i);
+    _decisionToDFA.emplace_back(_atn->getDecisionState(i), i);
   }
 }
 
