diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index 455539f..af3084b 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -31,8 +31,9 @@ if (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/highway/CMakeLists.txt" AND
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/highway/LICENSE"
                  ${PROJECT_BINARY_DIR}/LICENSE.highway COPYONLY)
 else()
-  find_package(HWY 1.0.7)
-  if (NOT HWY_FOUND)
+  find_package(HWY NAMES hwy CONFIG REQUIRED)
+  add_library(hwy ALIAS hwy::hwy)
+  if (NOT TARGET hwy::hwy)
     message(FATAL_ERROR
         "Highway library (hwy) not found. Install libhwy-dev or download it "
         "to third_party/highway from https://github.com/google/highway . "
