diff --git a/lib/jpegli_cms.cmake b/lib/jpegli_cms.cmake
index 5c13816..e8206ab 100644
--- a/lib/jpegli_cms.cmake
+++ b/lib/jpegli_cms.cmake
@@ -71,10 +71,14 @@ install(TARGETS jpegli_cms
 
 if (BUILD_SHARED_LIBS)
   set(JPEGLI_REQUIRES_TYPE "Requires.private")
+  if (NOT MSVC)
   set(JPEGLI_CMS_PRIVATE_LIBS "-lm ${PKGCONFIG_CXX_LIB}")
+  endif()
 else()
   set(JPEGLI_REQUIRES_TYPE "Requires")
+  if (NOT MSVC)
   set(JPEGLI_CMS_PRIVATE_LIBS "-lm ${PKGCONFIG_CXX_LIB}")
+  endif()
 endif()
 
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/cms/libjpegli_cms.pc.in"
