--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,6 +5,7 @@
 )
 
 include(CheckCXXCompilerFlag)
+include(CTest)
 include(CheckIPOSupported)
 
 check_cxx_compiler_flag("-std=c++26" COMPILER_SUPPORTS_CXX26)
@@ -35,6 +36,7 @@
 endif ()
 
 option(STATIC "static binary" OFF)
+option(GSHEETPP_BUILD_EXAMPLE "Build gsheetpp example" ON)
 if (CMAKE_COMPILER_IS_GNUCXX AND STATIC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -static -static-libstdc++ -static-libgcc")
 endif ()
@@ -118,9 +120,12 @@
   DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/gsheetpp
 )
 
-add_executable(gsheetpp_example ${CMAKE_SOURCE_DIR}/examples/main.cpp)
-target_link_libraries(gsheetpp_example PRIVATE gsheetpp)
-target_compile_features(gsheetpp_example PRIVATE ${STD_CPP})
+if (GSHEETPP_BUILD_EXAMPLE)
+  add_executable(gsheetpp_example ${CMAKE_SOURCE_DIR}/examples/main.cpp)
+  target_link_libraries(gsheetpp_example PRIVATE gsheetpp)
+  target_compile_features(gsheetpp_example PRIVATE ${STD_CPP})
+endif ()
 
-enable_testing()
-add_subdirectory(test)
+if (BUILD_TESTING)
+  add_subdirectory(test)
+endif ()
