diff --git a/source/MRMesh/MRMeshFwd.h b/source/MRMesh/MRMeshFwd.h
index 8c4f076..ad15fff 100644
--- a/source/MRMesh/MRMeshFwd.h
+++ b/source/MRMesh/MRMeshFwd.h
@@ -18,7 +18,7 @@
 // additionally MR_ITERATOR_DEBUG_LEVEL with the same value to indicate that it is done intentionally
 // (and you are ok with up to 100x slowdown).
 //
-#if defined _MSC_VER
+#if 0
     #if !defined _ITERATOR_DEBUG_LEVEL
         #define _ITERATOR_DEBUG_LEVEL 0
     #endif
