diff --git a/src/sparsehash/sparsetable b/src/sparsehash/sparsetable
index 6259ebd..4eb4e34 100644
--- a/src/sparsehash/sparsetable
+++ b/src/sparsehash/sparsetable
@@ -802,16 +802,17 @@ class destructive_two_d_iterator {
 template <class T, u_int16_t GROUP_SIZE, class Alloc>
 class sparsegroup {
  private:
-  typedef typename Alloc::template rebind<T>::other value_alloc_type;
+  typedef typename std::allocator_traits<Alloc>::template rebind_alloc<T> value_alloc_type;
+  typedef std::allocator_traits<value_alloc_type> value_alloc_traits;
 
  public:
   // Basic types
   typedef T value_type;
   typedef Alloc allocator_type;
-  typedef typename value_alloc_type::reference reference;
-  typedef typename value_alloc_type::const_reference const_reference;
-  typedef typename value_alloc_type::pointer pointer;
-  typedef typename value_alloc_type::const_pointer const_pointer;
+  typedef T& reference;
+  typedef const T& const_reference;
+  typedef typename value_alloc_traits::pointer pointer;
+  typedef typename value_alloc_traits::const_pointer const_pointer;
 
   typedef table_iterator<sparsegroup<T, GROUP_SIZE, Alloc> > iterator;
   typedef const_table_iterator<sparsegroup<T, GROUP_SIZE, Alloc> >
@@ -1289,8 +1290,8 @@ class sparsegroup {
   template <class A>
   class alloc_impl : public A {
    public:
-    typedef typename A::pointer pointer;
-    typedef typename A::size_type size_type;
+    typedef typename std::allocator_traits<A>::pointer pointer;
+    typedef typename std::allocator_traits<A>::size_type size_type;
 
     // Convert a normal allocator to one that has realloc_or_die()
     alloc_impl(const A& a) : A(a) { }
@@ -1311,8 +1312,8 @@ class sparsegroup {
   class alloc_impl<libc_allocator_with_realloc<A> >
       : public libc_allocator_with_realloc<A> {
    public:
-    typedef typename libc_allocator_with_realloc<A>::pointer pointer;
-    typedef typename libc_allocator_with_realloc<A>::size_type size_type;
+    typedef typename std::allocator_traits<libc_allocator_with_realloc<A>>::pointer pointer;
+    typedef typename std::allocator_traits<libc_allocator_with_realloc<A>>::size_type size_type;
 
     alloc_impl(const libc_allocator_with_realloc<A>& a)
         : libc_allocator_with_realloc<A>(a) { }
@@ -1362,20 +1363,21 @@ template <class T, u_int16_t GROUP_SIZE = DEFAULT_SPARSEGROUP_SIZE,
           class Alloc = libc_allocator_with_realloc<T> >
 class sparsetable {
  private:
-  typedef typename Alloc::template rebind<T>::other value_alloc_type;
-  typedef typename Alloc::template rebind<
-      sparsegroup<T, GROUP_SIZE, value_alloc_type> >::other vector_alloc;
+  typedef typename std::allocator_traits<Alloc>::template rebind_alloc<T> value_alloc_type;
+  typedef std::allocator_traits<value_alloc_type> value_alloc_traits;
+  typedef typename std::allocator_traits<Alloc>::template rebind_alloc<
+      sparsegroup<T, GROUP_SIZE, value_alloc_type> > vector_alloc;
 
  public:
   // Basic types
   typedef T value_type;                        // stolen from stl_vector.h
   typedef Alloc allocator_type;
-  typedef typename value_alloc_type::size_type size_type;
-  typedef typename value_alloc_type::difference_type difference_type;
-  typedef typename value_alloc_type::reference reference;
-  typedef typename value_alloc_type::const_reference const_reference;
-  typedef typename value_alloc_type::pointer pointer;
-  typedef typename value_alloc_type::const_pointer const_pointer;
+  typedef typename value_alloc_traits::size_type size_type;
+  typedef typename value_alloc_traits::difference_type difference_type;
+  typedef value_type& reference;
+  typedef const value_type& const_reference;
+  typedef typename value_alloc_traits::pointer pointer;
+  typedef typename value_alloc_traits::const_pointer const_pointer;
   typedef table_iterator<sparsetable<T, GROUP_SIZE, Alloc> > iterator;
   typedef const_table_iterator<sparsetable<T, GROUP_SIZE, Alloc> >
       const_iterator;
